Turtlesim shared lib

Dale-Koenig opened the pull request:
Reorganizing and reattempting https://github.com/rapyuta-robotics/alica/pull/502, with upcoming placeholders feature in mind.

Base package alica_turtlesim with all base plans.  Ros turtlesim package adds a couple new libraries and plugs them in.  Both are loaded and combined to get the full plan.

bjoernschroeder committed:
add turtle_interface package, add shared lib package for turtlesims

bjoernschroeder committed:
update ros1 turtlesim to use shared lib

bjoernschroeder committed:
update ros2 turtlesim to use shared lib

bjoernschroeder committed:
format

bjoernschroeder committed:
add gdb to ros2 turtles

bjoernschroeder committed:
move libalica-ros-utils behaviours to libalica-turtlesim, add subscribers for WaitForMsg and WaitForTrigger

bjoernschroeder committed:
remove logs, update ros1 interface

bjoernschroeder committed:
address review comments for ros1 turtlesim

bjoernschroeder committed:
address review comments for ros2

bjoernschroeder committed:
remove ros2 changes

bjoernschroeder committed:
share etc foldere for ros1

bjoernschroeder committed:
remove old code

bjoernschroeder committed:
format

bjoernschroeder committed:
chagne logging verbosity to info

Commit:
add back log, remove unused behaviours in libalica-turtlesim

Commit:
add back missing spawn logs

Commit:
create alica_turtlesim subfolder with shared plans, configs and turtle interface base class

Commit:
remove parent include folder from cmake

Commit:
format

Commit:
add back logs

Commit:
remove old ament cmake requirement

Commit:
initialize member _name in initializer list

Commit:
fix includes

Commit:
use inherited _name member variable

Commit:
pass paths as vector

bjoernschroeder committed:
update ros2 ci workflow

bjoernschroeder committed:
skip building libalica-turtlesim for ros2

Commit:
add back failure log

Commit:
add comment for rotate method

Commit:
add comment for rotate method

Commit:
Reorg

Commit:
Working in (only) install

Commit:
Allow finding as catkin package

Commit:
Fix to work both in/out of install

Commit:
Maybe fix some includes with build order

bjoernschroeder commented:
Is it in a runnable state? When I try to run tests it says it cannot find the master plan

Commit:
fix launch file for test

veerajsk committed:
Fix crash: dll access after its destruction (#514)

The dynamic creators maintain a reference to the currently loaded
symbol, therefore dlls that were loaded prior would no longer be loaded
& accessing such dlls (typically in the destructor of the beh/plan etc)
would cause undefined behaviour. Specifically this lead to a crash in
the destructor of the GoTo behaviour in turtlesim. Such crashes are
hard to debug & can occur in more ways due to this issue.
As a fix, the creator for each symbol is stored for the lifetime of
the dynamic creator object itself which is basically the entire
duration of the engine. Consequently the symbol & dll is loaded
just once potentially improving performance.

Commit:
Fix missing include

Dale-Koenig commented:
@bjoernschroeder @veerajsk seems everything works now

veerajsk committed:
Merge branch 'devel' into dk/turtlesim_shared_lib

bjoernschroeder commented on supplementary/alica_ros1/alica_ros_turtlesim/src/turtle_ros1_interfaces.cpp:
```diff
@@ -1,24 +1,28 @@
-#include "turtle_interfaces.hpp"
+#include "alica_ros_turtlesim/turtle_ros1_interfaces.hpp"
+
+#include <engine/blackboard/Blackboard.h>
```
seems like its not used
